### ISSUE TITLE
NPE in AbstractLanguageServerTest with WorkspaceEdit

### ISSUE THREAD

**Issue by dunnry** [eclipse/xtext-core#1380]
ALST's `toExpectation` method for `WorkspaceEdit` has a non-null checked call to its `changes` collection.  Depending on how you create your WorkspaceEdit, you will trigger this error:

```
//no error because default ctor initializes changes collection
val edit = WorkspaceEdit()
edit.documentChanges = ...

//this will cause NPE on toExpectation check
val edit2 = WorkspaceEdit(documentChanges)

```
The fix is that `WorkspaceEdit` should initialize the `changes` collection in all cases, or that `toExpectation `should check for null.

**Comment by cdietrich** [comment]
Hi, would you come up with a fix?

**Comment by cdietrich** [comment]
@dunnry which Xtext/LSP4j version do you use.

lsp4j >= 0.8.1 creates

```
new() {
		this.changes = new LinkedHashMap
	}
```

and thus this is not a problem

**Comment by dunnry** [comment]
I think we are using 0.8.0, but I see this still an issue in 0.8.1 as well (unless I am reading Java wrong):

```
 public WorkspaceEdit() {
    LinkedHashMap<String, List<TextEdit>> _linkedHashMap = new LinkedHashMap<String, List<TextEdit>>();
    this.changes = _linkedHashMap;
  }
  
  public WorkspaceEdit(final Map<String, List<TextEdit>> changes) {
    this.changes = changes;
  }
  
  public WorkspaceEdit(final List<Either<TextDocumentEdit, ResourceOperation>> documentChanges) {
    this.documentChanges = documentChanges; //<== this.changes is null here.
  }
```

'this.changes' is left uninitialized in the case of the 3rd ctor.  Since this ctor is the one that most people use now (`List<TextEdit>` one should be deprecated), it is causing the NPE when used with the `AbstractLanguageServer` test that tries to enumerate the `changes` collection.  I didn't check, but I would assume you would also have an NPE the other way (where you supplied the `changes` ctor and didn't use the `documentChanges` one).

**Comment by dunnry** [comment]
I suspect that because `WorkspaceEdit` can contain either `changes` or `documentChanges`, but not both that the correct resolution would be to update the `toExpectation` method.  I will try to tackle that later when I get a chance and get my Eclipse contribution license, etc., etc.

**Comment by cdietrich** [comment]
ahhh got you. just had a look at the empty constructor

**Comment by cdietrich** [comment]
Fixed in 2.21. Thx @dunnry
